dataland-frontend/tests/component/components/resources/frameworkDataSearch/ThreeLayerDataTable.cy.ts

```diff
@@ -0,0 +1,147 @@
+    cy.mountWithPlugins(ThreeLayerDataTable, {
+      keycloak: minimalKeycloakMock({}),
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      props: {
+        dataModel: euTaxonomyForNonFinancialsDisplayDataModel,
+        dataAndMetaInfo: mockedDataForTest,
+      },
+    }).then(() => {
+      cy.get("[data-test='TwoLayerTest']")
+        .get(".d-table-style")
+        .each((element, index) => {
+          cy.wrap(element).eq(0).eq(0).get(".p-badge").eq(index).should("have.text", expectedOrderOfCategories[index]);
+        });
+
+      toggleCategoryByClick(dataTestTagsOfCategories[0]);
+
+      /**
+       * The goal for the loop is to expand one KPI at a time and check the order of the entries.
+       */
+
+      let subcategoriesForCurrentCategory;
+      for (let categoryIndex = 1; categoryIndex < dataTestTagsOfCategories.length; categoryIndex++) {
+        subcategoriesForCurrentCategory = dataTestTagsOfSubcategoriesGroupedByCategories[categoryIndex];
+
+        toggleCategoryByClick(dataTestTagsOfCategories[categoryIndex]);
+
+        for (let subCategoryIndex = 0; subCategoryIndex < subcategoriesForCurrentCategory.length; subCategoryIndex++) {
+          cy.get(".p-rowgroup-header")
+            .filter(":visible")
+            .eq(subCategoryIndex)
+            .get(`span[id="${subcategoriesForCurrentCategory[subCategoryIndex]}"]`)
+            .should("contain", `${expectedOrderOfSubcategoriesGroupedByCategories[categoryIndex][subCategoryIndex]}`);
+        }
+        toggleCategoryByClick(dataTestTagsOfCategories[categoryIndex]);
+      }
+    });
+  });
+
+  it("Opens the aligned activities modal and checks that it works as intended", () => {
+    cy.mountWithDialog(
+      ThreeLayerDataTable,
+      {
+        keycloak: minimalKeycloakMock({}),
+      },
+      {
+        dataModel: euTaxonomyForNonFinancialsDisplayDataModel,
+        dataAndMetaInfo: mockedDataForTest,
+        modalColumnHeaders: euTaxonomyForNonFinancialsModalColumnHeaders,
+        sortBySubcategoryKey: false,
+      },
+    ).then(() => {
+      /**
+      toggleCategoryByClick("Basic Information");
+      toggleCategoryByClick("CapEx");
+      cy.get(`[data-test='totalAlignedShare']`).filter(":visible").click();
+      cy.get(`[data-test='totalAlignedShare']`)
+        .filter(":visible")
+        .get("em")
+        .filter(":visible")
+        .eq(-1)
+        .should("have.text", " dataset ")
+        .click();
+      cy.wait(10000);
+      cy.get("table").find(`tr:contains("Activity")`);
+      cy.get("table").find(`tr:contains("Code(s)")`);
+      cy.get("table").find(`tr:contains("Revenue")`);
+      cy.get("table").find(`tr:contains("Climate change mitigation")`);
+      cy.get("table").find(`tr:contains("Climate change adaptation")`);
+      cy.get("table").find(`tr:contains("Water and marine resources")`);
+      cy.get("table").find(`tr:contains("Circular economy")`); *
+       */
```

Comment:
Why is all of this commented out? Currently the test does not do anything right?

Reply:
Test is WiP by @marcelrasbachdfine

Reply:
No longer commented out